src/optimagic/optimizers/pyswarms_optimizers.py

```diff
@@ -0,0 +1,913 @@
+
+    p_norm: Literal[1, 2]
+    """Distance metric for neighbor selection (1=Manhattan, 2=Euclidean)."""
+
+
+@mark.minimizer(
+    name="pyswarms_global_best",
+    solver_type=AggregationLevel.SCALAR,
+    is_available=IS_PYSWARMS_INSTALLED,
+    is_global=True,
+    needs_jac=False,
+    needs_hess=False,
+    needs_bounds=False,
+    supports_parallelism=True,
+    supports_bounds=True,
+    supports_infinite_bounds=False,
+    supports_linear_constraints=False,
+    supports_nonlinear_constraints=False,
+    disable_history=False,
+)
+@dataclass(frozen=True)
+class PySwarmsGlobalBestPSO(Algorithm):
+    r"""Minimize a scalar function using Global Best Particle Swarm Optimization.
+
+    A population-based stochastic, global optimization optimization algorithm that
+    simulates the social behavior of bird flocking or fish schooling. Particles
+    (candidate solutions) move through the search space, adjusting their positions
+    based on their own experience (cognitive component) and the experience of their
+    neighbors or the entire swarm (social component).
+
+    This implementation uses a star topology where all particles are connected to
+    each other, making each particle aware of the global best solution found by the
+    entire swarm.
+
+    The position update follows:
+
+    .. math::
+
+        x_{i}(t+1) = x_{i}(t) + v_{i}(t+1)
+
+    The velocity update follows:
+
+    .. math::
+
+        v_{ij}(t+1) = w \cdot v_{ij}(t) + c_1 r_{1j}(t)[y_{ij}(t) - x_{ij}(t)]
+                      + c_2 r_{2j}(t)[\hat{y}_j(t) - x_{ij}(t)]
+
+    Where:
+        - :math:`w`: inertia weight controlling momentum
+        - :math:`c_1`: cognitive parameter for attraction to personal best
+        - :math:`c_2`: social parameter for attraction to global best
+        - :math:`r_{1j}, r_{2j}`: random numbers in [0,1]
+        - :math:`y_{ij}(t)`: personal best position of particle i
+        - :math:`\hat{y}_j(t)`: global best position
+
+    This algorithm is an adaptation of the original Particle Swarm Optimization method
+    by :cite:`Kennedy1995`
+
+    """
+
+    n_particles: PositiveInt = 50
+    """Number of particles in the swarm."""
+
+    cognitive_parameter: PositiveFloat = 0.5
+    r"""Cognitive parameter :math:`c_1` controlling attraction to personal best."""
+
+    social_parameter: PositiveFloat = 0.3
+    r"""Social parameter :math:`c_2` controlling attraction to global best."""
+
+    inertia_weight: PositiveFloat = 0.9
+    r"""Inertia weight :math:`w` controlling momentum."""
+
+    convergence_ftol_rel: NonNegativeFloat = CONVERGENCE_FTOL_REL
+    """Stop when relative change in objective function is less than this value."""
+
+    convergence_ftol_iter: PositiveInt = 1
+    """Number of iterations to check for convergence."""
+
+    stopping_maxiter: PositiveInt = STOPPING_MAXITER
+    """Maximum number of iterations."""
+
+    initial_population: NDArray[np.float64] | None = None
```

Comment:
`list[PyTree]`, then use the converter to convert this to a list of arrays and then a 2d array from there.